tortoise/fields/base.py

```diff
@@ -441,3 +463,21 @@ def default_name(default: Any) -> Optional[Union[int, float, str, bool]]:
             desc["db_field_types"] = self.get_db_field_types()
 
         return desc
+
+    def describe_by_dataclass(self) -> FieldDescriptionBase:
```

Comment:
Disclaimer: I'm quite new to the project but I looked into `pydantic_model_creator` a bit and it's obvious that it's convoluted, it has issues and begs for refactoring. So thank you for working on this!

I'm not sure though that extending the core classes like `Field`, `RelationalField` and others with functionality required for one particular extension is optimal. Basically `contrib.pydantic` is not a part of ORM, it's an extension that only some users use. So it seems to me that it would make more sense to have the `describe_by_dataclass` logic inside of `contrib.pydantic`.

Reply:
That is a good point. Originally I planned to change the describe() method to use the describe_by_dataclass() method, with something like asdict(self.describe_by_dataclass()). That is because describe() only returns a dict, so typing is not working with describe(). But this change would become a bit more larger and out of focus of the original intention of this PR. with other words: I think describe_by_dataclass() could be helpfull in many points, but I can see why it should be moved to a dedicated place.

Reply:
moved to tortoise.contrib.pydantic.dataclasses